### PR TITLE
Connect/read blobstore timeouts enabled only for API server

### PR DESCRIPTION
**Pull request by xbrianh**
* Use default client timeouts for daemons, which are unconstrained by API Gateway.
* Revert timeout changes to `tests/test_gscopyclient.py`

### CODE REVIEW

File: dss/config.py [hunk 1]
```diff
@@ -116,32 +116,42 @@ def set_config(config: BucketConfig):
...
+    def get_native_handle(replica: "Replica") -> typing.Any:
         if replica == Replica.aws:
+            return Config._get_native_aws_handle()
+        elif replica == Replica.gcp:
+            return Config._get_native_gcp_handle()
+        raise NotImplementedError(f"Replica `{replica.name}` is not implemented!")
+
+    @staticmethod
+    def _get_native_aws_handle() -> typing.Any:
+        if Config.BLOBSTORE_CONNECT_TIMEOUT is None or Config.BLOBSTORE_READ_TIMEOUT is None:
```
Comment (ttung) [inline dss/config.py]: what if `BLOBSTORE_BOTO_RETRIES` is set?

Comment (ttung) [reply dss/config.py]: Also, this is an _or_, which is kind of weird.

File: dss/config.py [hunk 2]
```diff
@@ -116,32 +116,42 @@ def set_config(config: BucketConfig):
...
                 )
             )
-        elif replica == Replica.gcp:
+
+    @staticmethod
+    def _get_native_gcp_handle() -> typing.Any:
+        if Config.BLOBSTORE_GS_MAX_CUMULATIVE_RETRY is not None:
             google.resumable_media.common.MAX_CUMULATIVE_RETRY = Config.BLOBSTORE_GS_MAX_CUMULATIVE_RETRY
 
+        if Config.BLOBSTORE_CONNECT_TIMEOUT is None or Config.BLOBSTORE_READ_TIMEOUT is None:
```
Comment (ttung) [inline dss/config.py]: ditto, also an _or_.

File: dss/config.py (hunk 1 above)
Comment (xbrianh) [reply dss/config.py]: My idea here was that you should either provide values for all timeouts, or none of them.

Switched to allowing partial configs.

File: chalice/app.py [hunk 3]
```diff
@@ -28,6 +28,10 @@
 configure_lambda_logging()
 
 Config.set_config(BucketConfig.NORMAL)
+Config.BLOBSTORE_CONNECT_TIMEOUT = 5
```
Comment (hannes-ucsc) [inline chalice/app.py]: This is why I think the upper-case convention is useless, and actually misleading. Here you're reassigning to a constant. There are no constants in Python and we should pretend it does.

Anyhow, different PR.

Comment (ttung) [reply chalice/app.py]: Yeah, I agree that this is kind of weird, but it's unclear to me what the solution is.

**Comment by xbrianh** [comment]
connects to #1098

**Pull request merged**